src/jsoncharutils_tables.h

```diff
@@ -215,977 +215,651 @@ static const double power_of_ten[] = {
     1e0,  1e1,  1e2,  1e3,  1e4,  1e5,  1e6,  1e7,  1e8,  1e9,  1e10, 1e11,
     1e12, 1e13, 1e14, 1e15, 1e16, 1e17, 1e18, 1e19, 1e20, 1e21, 1e22};
 
-// the mantissas of powers of ten from -308 to 308, extended out to sixty four
-// bits
-// This struct will likely get padded to 16 bytes.
-typedef struct {
-  uint64_t mantissa;
-  int32_t exp;
```

Comment:
So removing this (which takes 64 bits) is where you get the savings. Twice as much in cache, nice!

Reply:
Yes. The int32_t exp took 64 bits in practice  because the struct ended up padded (this is not specified by the standard, but that's what I measured on all compilers...

```C++
struct joe {
  uint64_t x;
  uint32_t y;
};

// implies sizeof(joe) == 16;

```

So this was very fat code. I was happy to leave it there because I thought that computing floor(log(5**x)/log(2)) would be too expensive... but as you can see, it can be done with a multiplication and a shift... which is not more expensive than a lookup.

Reply:
And yes, because this array is accessed a lot, compacting it by a factor of two will reduce cache usage by a factor of two. So it should be a win.

It is not going to show up in our benchmarks because the data is already in cache in our tests... but in the real world, this will be a win.